opni-insights-service/app/main.py

```diff
@@ -473,17 +475,31 @@ async def get_logs(start_ts, end_ts):
         ],
         "sort": [{"timestamp": {"order": "asc"}}],
     }
-    try:
-        async for each_result in async_scan(
-            es_instance, index="logs", query=query_body
-        ):
-            logs_dict["Logs"].append(each_result["_source"])
-    except Exception as e:
-        logging.error(f"Unable to access logs data. {e}")
-        return logs_dict
-
-    return logs_dict
-
+    scroll_value = "1m"
+    # If scroll_id is provided, then use it to fetch the next 100 logs and return that in addition to the updated scroll_id as part of the logs_dict dictionary.
+    if scroll_id:
```

Comment:
I think codes in if/else block are almost the same (except the first line `current_page = xxx`). let's merge them.